src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book alphabetically as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD;
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+        List<ReadOnlyPerson> mutablePersons = new ArrayList<>(allPersons);
+        Collections.sort(mutablePersons, new SortByName());
+        return new CommandResult(getMessageForPersonListShownSummary(mutablePersons), mutablePersons);
+    }
+}
+
+class SortByName implements Comparator<ReadOnlyPerson> {
```

Comment:
Should this class be stored in a separate file?

Reply:
@Articho28 Given this not a public class, does it require a seperate file? Nevertheless, @daviddalmaso should this be a class or a method?